addons/survey/models/survey_user.py

```diff
@@ -276,6 +281,29 @@ def _get_line_comment_values(self, question, comment):
             'value_char_box': comment,
         }
 
+    def _get_answer_score_values(self, question, answer, answer_type):
+        # default and non-scored questions
+        vals = {'answer_score': 0, 'answer_is_correct': False}
+        # record selected suggested choice assigned score (can be: pos, neg, or 0)
+        if answer_type in ['suggestion']:
+            if answer:
+                vals.update({'answer_score': self.env['survey.question.answer'].browse(int(answer)).answer_score,
+                             'answer_is_correct': self.env['survey.question.answer'].browse(int(answer)).is_correct})
+        # record assigned score only when correct in all other scored question cases (can be: pos or 0)
+        elif self.env['survey.question'].browse(int(question.id)).is_scored_question:
+            answer_score = self.env['survey.question'].browse(int(question.id)).answer_score
+            if answer_type == 'numerical_box':
+                answer = float(answer)
+            elif answer_type == 'date':
+                answer = fields.Date.from_string(answer)
+            elif answer_type == 'datetime':
+                answer = fields.Datetime.from_string(answer)
+            if answer == self.env['survey.question'].browse(int(question.id))._get_answer():
```

Comment:
Why do you re-browse survey.question multiple times here?
Where you could simply do `question.is_scored_question`, `question.answer_score` and `question._get_answer()` ?

Reply:
Oops, was too hasty with copy/paste. Fixed it.